src/otx/algo/object_detection_3d/detectors/monodetr.py

```diff
@@ -0,0 +1,313 @@
+                        nn.GroupNorm(32, hidden_dim),
+                    ),
+                )
+                in_channels = hidden_dim
+            self.input_proj = nn.ModuleList(input_proj_list)
+        else:
+            self.input_proj = nn.ModuleList(
+                [
+                    nn.Sequential(
+                        nn.Conv2d(backbone.num_channels[0], hidden_dim, kernel_size=1),
+                        nn.GroupNorm(32, hidden_dim),
+                    ),
+                ],
+            )
+
+        self.backbone = backbone
+        self.aux_loss = aux_loss
+        self.with_box_refine = with_box_refine
+        self.num_classes = num_classes
+
+        for proj in self.input_proj:
+            nn.init.xavier_uniform_(proj[0].weight, gain=1)
+            nn.init.constant_(proj[0].bias, 0)
+        # if two-stage, the last class_embed and bbox_embed is for region proposal generation
+        num_pred = depthaware_transformer.decoder.num_layers
+        if with_box_refine:
+            self.class_embed = get_clones(self.class_embed, num_pred)
+            self.bbox_embed = get_clones(self.bbox_embed, num_pred)
+            nn.init.constant_(self.bbox_embed[0].layers[-1].bias.data[2:], -2.0)
+            # implementation for iterative bounding box refinement
+            self.depthaware_transformer.decoder.bbox_embed = self.bbox_embed
+            self.dim_embed_3d = get_clones(self.dim_embed_3d, num_pred)
+            self.depthaware_transformer.decoder.dim_embed = self.dim_embed_3d
+            self.angle_embed = get_clones(self.angle_embed, num_pred)
+            self.depth_embed = get_clones(self.depth_embed, num_pred)
+        else:
+            nn.init.constant_(self.bbox_embed.layers[-1].bias.data[2:], -2.0)
+            self.class_embed = nn.ModuleList([self.class_embed for _ in range(num_pred)])
+            self.bbox_embed = nn.ModuleList([self.bbox_embed for _ in range(num_pred)])
+            self.dim_embed_3d = nn.ModuleList([self.dim_embed_3d for _ in range(num_pred)])
+            self.angle_embed = nn.ModuleList([self.angle_embed for _ in range(num_pred)])
+            self.depth_embed = nn.ModuleList([self.depth_embed for _ in range(num_pred)])
+            self.depthaware_transformer.decoder.bbox_embed = None
+
+    def forward(
+        self,
+        images: Tensor,
+        calibs: Tensor,
+        img_sizes: Tensor,
+        targets: list[dict[str, Tensor]] | None = None,
+        mode: str = "predict",
+    ) -> dict[str, Tensor]:
+        """Forward method of the MonoDETR model.
+
+        Args:
+            images (list[Tensor]): images for each sample
+            calibs (Tensor): camera matrices for each sample
+            img_sizes (Tensor): image sizes for each sample
+            targets (list[dict[Tensor]): ground truth boxes and labels for each
+                sample
+            mode (str): The mode of operation. Defaults to "predict".
+        """
```

Comment:
Could you add docstring for return value?

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/4031/commits/00200f4d4811d82e86e371cb5dabd2541098e83c